salt/orch/kubernetes.sls

```diff
@@ -108,12 +112,7 @@ etcd_setup:
     - tgt_type: grain_pcre
     - sls:
       - etcd
-    # Currently, we never ask for more than 3 members. Setting this to 3 ensures
-    # we don't let more than 3 members attempt etcd discovery before a cluster
-    # has been fully formed. If we have less this 3, this will still succeed, as
-    # the exact number of members we expect will also end up attempting discovery
-    # at the same time.
-    - batch: 3
+    - batch: {{ salt.caasp_filters.get_max([default_batch, num_etcd_masters]) }}
```

Comment:
This will work reliably - we added the 3 hardcode to ensure that e.g. 5 nodes didn't ask "Can I be a master?" at once. When that happened - we often found more than the expected number believed they got a "yes" answer.

We want to run this state in batches of exactly the number of etc members we're expecting, otherwise, we'll just start hitting that issue again.